openpgp/packet/signature.go

```diff
@@ -841,7 +841,7 @@ func (sig *Signature) buildSubpackets(issuer PublicKey) (subpackets []outputSubp
 	}
 	if sig.IssuerFingerprint != nil {
 		contents := append([]uint8{uint8(issuer.Version)}, sig.IssuerFingerprint...)
-		subpackets = append(subpackets, outputSubpacket{true, issuerFingerprintSubpacket, true, contents})
+		subpackets = append(subpackets, outputSubpacket{true, issuerFingerprintSubpacket, sig.V5Keys, contents})
```

Comment:
This is a bit subtle but `sig.V5Keys` means "does this software support V5 keys", not "is this a V5 signature". So this should probably be (untested):
```suggestion
		subpackets = append(subpackets, outputSubpacket{true, issuerFingerprintSubpacket, sig.Version == 5, contents})
```